File: csharp_scripts/MainAppManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;
public class MainAppManager : MonoBehaviour
{
    public FixedIntensityProtocolManager fixedIntensityProtocolManager;
    public VariableIntensityProtocolManager variableIntensityProtocolManager;
    public PIPRMonocolar pIPRMonocolar;
    public ExtendedPIPRMonocular extendedPIPRMonocular;
    public DiameterSmoothingManger diameterSmoothingManger;

    public ResultConvertJsonObj resultConvertJsonObj;


    public enum EyeProtocolType
    {
        FixedIntensity_Bino,
        VariableIntensity_Bino,
        QuickTest_Bino,
        ExtendedPIPR_Bino,
        PIPR_Mono,
        ExtendedPIPR_Mono
    }

    public EyeProtocolType currentProtocol;

    // Start is called before the first frame update
    void Start()
    {
       // currentProtocol = EyeProtocolType.FixedIntensity_Bino;
       // Invoke("GetProtocolData", 1f);

    }


    public void GetProtocolData()
    {
        switch (currentProtocol)
        {
            case EyeProtocolType.ExtendedPIPR_Bino:

                break;
            case EyeProtocolType.FixedIntensity_Bino:

                 fixedIntensityProtocolManager.FixedIntensityTestBinocular();
            //    resultConvertJsonObj.ShowPupilXResult();

                 // pIPRMonocolar.PIPRMonocularTest();
                 // variableIntensityProtocolManager.VariableIntensityTestBinocular();

                break;
            case EyeProtocolType.QuickTest_Bino:
              
                break;
            case EyeProtocolType.VariableIntensity_Bino:

                variableIntensityProtocolManager.VariableIntensityTestBinocular();

                break;
            case EyeProtocolType.PIPR_Mono:
                pIPRMonocolar.PIPRMonocularTest();
                break;
            case EyeProtocolType.ExtendedPIPR_Mono:
                extendedPIPRMonocular.ExtendedPIPRMonocularTest();

                break;

        }
    }

}

File: app/src/main/java/com/example/pupilx/managers/MainAppManager.kt
package com.example.pupilx.managers

import com.example.pupilx.models.PatientResultData

class MainAppManager {
    private val fixedIntensityProtocolManager = FixedIntensityProtocolManager()
    private val variableIntensityProtocolManager = VariableIntensityProtocolManager()
    private val piprMonocular = PIPRMonocular()
    private val extendedPIPRMonocular = ExtendedPIPRMonocular()
    private val emergencyCareProtocolMonocular = EmergencyCareProtocolMonocular()
    private val whiteLightMonocular = WhiteLightMonocular()
    
    private val androidDataManager = AndroidDataManager()
    
    var patientResultData: PatientResultData = PatientResultData()
    
    enum class EyeProtocolType {
        FixedIntensity_Bino,
        VariableIntensity_Bino,
        QuickTest_Bino,
        ExtendedPIPR_Bino,
        PIPR_Mono,
        ExtendedPIPR_Mono,
        EmergencyCare_Mono,
        WhiteLight_Mono
    }
    
    var currentProtocol: EyeProtocolType = EyeProtocolType.FixedIntensity_Bino
    
    fun getProtocolData() {
        // Clear previous results
        patientResultData.velocityCalculations.clear()
        patientResultData.graphValueCalculation.clear()
        patientResultData.graphValueCalculation.clear()
        
        when (currentProtocol) {
            EyeProtocolType.ExtendedPIPR_Bino -> {
                // Implementation would go here
            }
            EyeProtocolType.FixedIntensity_Bino -> {
                fixedIntensityProtocolManager.fixedIntensityTestBinocular()
            }
            EyeProtocolType.QuickTest_Bino -> {
                // Implementation would go here
            }
            EyeProtocolType.VariableIntensity_Bino -> {
                variableIntensityProtocolManager.variableIntensityTestBinocular()
            }
            EyeProtocolType.PIPR_Mono -> {
                piprMonocular.piprMonocularTest()
            }
            EyeProtocolType.ExtendedPIPR_Mono -> {
                extendedPIPRMonocular.extendedPIPRMonocularTest()
            }
            EyeProtocolType.EmergencyCare_Mono -> {
                emergencyCareProtocolMonocular.emergencyCareProtocolTest()
            }
            EyeProtocolType.WhiteLight_Mono -> {
                whiteLightMonocular.whiteLightPIPRTest()
            }
        }
    }
    
    fun saveResultsToFile(filePath: String) {
        // Implementation would go here
        // This would use the JsonUtils to save the results
    }
}